libs/payments/cart/src/lib/cart.service.ts

```diff
@@ -257,6 +257,26 @@ export class CartService {
     }
   }
 
+  /**
+   * Fetch couponCode from a cart, validating the cart version.
+   */
+  @SanitizeExceptions()
+  async getCoupon(args: {
+    cartId: string;
+    version: number;
+  }): Promise<string | null> {
+    const cart = await this.cartManager.fetchCartById(args.cartId);
+
+    if (cart.version != args.version) {
+      throw new CartInvalidStateForActionError(
+          args.cartId,
+          cart.state,
+          'getCoupon'
+        );
+    }
+    return cart.couponCode;
```

Comment:
suggestion (non-blocking): Could this follow the pattern set by `getCartState` for returning data?